pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,627 @@
+		apiextensionClient:            apiExtensionClient,
+		eventRecorder:                 eventRecorder,
+		queue:                         workqueue.NewTypedRateLimitingQueueWithConfig(workqueue.DefaultTypedControllerRateLimiter[string](), workqueue.TypedRateLimitingQueueConfig[string]{Name: "TargetConfigReconciler"}),
+		targetImage:                   targetImage,
+		namespace:                     namespace,
+	}
+
+	operatorClientInformer.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c *TargetConfigReconciler) sync() error {
+	leaderWorkerSetOperator, err := c.operatorClient.Get(c.ctx, operatorclient.OperatorConfigName, metav1.GetOptions{})
+	if err != nil {
+		klog.ErrorS(err, "unable to get operator configuration", "namespace", c.namespace, "openshift-lws-operator", operatorclient.OperatorConfigName)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleManager(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage manager cluster role err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleMetrics(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage metrics cluster role err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleProxy(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage proxy cluster role err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleBindingManager(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage manager cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleBindingMetrics(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage metrics cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageClusterRoleBindingProxy(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage proxy cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRole(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRoleMonitoring(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRoleBinding(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageRoleBindingMonitoring(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage cluster role binding err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageConfigmap(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage configmap err: %v", err)
+		return err
+	}
+
+	_, _, err = c.manageCustomResourceDefinition(leaderWorkerSetOperator)
+	if err != nil {
+		klog.Errorf("unable to manage leaderworkerset CRD err: %v", err)
+		return err
+	}
+
+	deployment, _, err := c.manageDeployments(leaderWorkerSetOperator)
```

Comment:
no big impact, but IMO the deployment should be last behind e.g. the webhooks

Reply:
yes, moved to the end.